Backend/app/rider-platform/rider-app/Main/src/Domain/Action/UI/Booking.hs

```diff
@@ -21,28 +21,158 @@ where
 
 import Data.OpenApi (ToSchema (..))
 import qualified Domain.Types.Booking as SRB
+import qualified Domain.Types.Booking.BookingLocation as DBL
 import qualified Domain.Types.Person as Person
+import qualified Domain.Types.Ride as Ride
 import EulerHS.Prelude hiding (id)
 import Kernel.Storage.Esqueleto (runInReplica)
 import Kernel.Storage.Esqueleto.Config (EsqDBReplicaFlow)
 import Kernel.Types.Id
 import Kernel.Utils.Common
-import Storage.CachedQueries.CacheConfig (CacheFlow)
+import SharedLogic.Booking
+import qualified SharedLogic.SimulatedFlow.Booking as SSBooking
+import Storage.CachedQueries.CacheConfig (CacheFlow, SimluatedCacheFlow)
+import qualified Storage.CachedQueries.SimulatedFlow.Driver as CSD
+import Storage.CachedQueries.SimulatedFlow.SearchRequest
+import qualified Storage.CachedQueries.SimulatedFlow.SearchRequest as CSR
 import qualified Storage.Queries.Booking as QRB
+import qualified Storage.Queries.Person as QP
 import Tools.Error
 
 newtype BookingListRes = BookingListRes
   { list :: [SRB.BookingAPIEntity]
   }
   deriving (Generic, Show, FromJSON, ToJSON, ToSchema)
 
-bookingStatus :: (CacheFlow m r, EsqDBFlow m r, EsqDBReplicaFlow m r) => Id SRB.Booking -> Id Person.Person -> m SRB.BookingAPIEntity
+bookingStatus :: (CacheFlow m r, EsqDBFlow m r, EsqDBReplicaFlow m r, SimluatedCacheFlow m r) => Id SRB.Booking -> Id Person.Person -> m SRB.BookingAPIEntity
 bookingStatus bookingId personId = do
-  booking <- runInReplica (QRB.findById bookingId) >>= fromMaybeM (BookingDoesNotExist bookingId.getId)
-  unless (booking.riderId == personId) $ throwError AccessDenied
-  SRB.buildBookingAPIEntity booking
+  person <- runInReplica (QP.findById personId) >>= fromMaybeM (PersonNotFound personId.getId)
+  if person.isSimulated
```

Comment:
can you make interceptSimulationFlow functions for all the APIs where we are having this diversion ? also move the helper functions like `createBookingList` from `Domain...` modules to `SharedLogic.SimulationFlow....`